src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +116,341 @@ def execute(self, reqStatus):
+        the given workflow
+        :param  wflow:   A MSRuleCleaner workflow representation
+        :param  gran:    Data granularity to search for Rucio rules. Possible values:
+                        'block' || 'container'
+        :return:        wflow
+        """
+        currPline = wflow['PlineMarkers'][-1]
+        # Find all the output placement rules created by the agents
+        # NOTE: Check MSOutput 'TransferStatus' first and assure it is 'done'
+        #       only then we proceed with building the respective list of rules
+        if wflow['TransferDone']:
+            for dataCont in wflow['OutputDatasets']:
+                if gran == 'container':
+                    wflow['RulesToClean'][currPline].extend(self.rucio.listDataRules(dataCont, account=rucioAcct))
+                elif gran == 'block':
+                    blocks = self.rucio.getBlocksInContainer(dataCont)
+                    for block in blocks:
+                        wflow['RulesToClean'][currPline].extend(self.rucio.listDataRules(block, account=rucioAcct))
+        return wflow
+
+    def cleanRucioRules(self, wflow):
+        """
+        Cleans all the Rules present in the field 'RulesToClean' in the MSRuleCleaner
+        workflow representation. And fills the relevant Cleanup Status.
+        :param wflow:   A MSRuleCleaner workflow representation
+        :return:        wflow
+        """
+        # NOTE: The function should be called independently and sequentially from
+        #       The Input and the respective BlockLevel pipelines.
+
+        # NOTE: The current functional pipeline is always the last one in the PlineMarkers list
+        currPline = wflow['PlineMarkers'][-1]
+        delResutls = []
+        if wflow['TransferDone']:
+            if self.msConfig['enableRealMode']:
+                for rule in wflow['RulesToClean'][currPline]:
+                    msg = "%s: Deleting rule['id']: %s "
+                    msg += "For did: %s"
+                    self.logger.info(msg, currPline, rule['id'], rule['name'])
+                    delResult = self.rucio.deleteRule(rule['id'])
+                    delResutls.append(delResult)
+                    if not delResult:
+                        msg = "%s: Failed to delete rule['id']: %s "
+                        msg += "For did: %s"
+                        self.logger.warning(msg, currPline, rule['id'], rule['name'])
+            else:
+                for rule in wflow['RulesToClean'][currPline]:
+                    delResutls.append(True)
+                    msg = "%s: Is about to delete rule['id']: %s "
+                    msg += "For did: %s"
+                    self.logger.debug(msg, currPline, rule['id'], rule['name'])
+        else:
+            delResutls.append(False)
+            msg = "%s: 'TransferStatus' is 'pending' or 'TransferInfo' is missing. "
+            msg += "Skip deleting any rules for: %s "
+            self.logger.debug(msg, currPline, wflow['RequestName'])
```

Comment:
I think this log level can be made INFO.

Reply:
This message does not exist any more.